trulens_eval/trulens_eval/feedback.py

```diff
@@ -1,5 +1,353 @@
+  specification, for `statement` was of this type. The input to `aggregate` must
+  be a method which can be imported globally. This requirement is further
+  elaborated in the next section. This function is called on the `float` results
+  of feedback function evaluations to produce a single float. The default is
+  `numpy.mean`.
+
+The result of these lines is that `f_qs_relevance` can be now be run on
+app/records and will automatically select the specified components of those
+apps/records:
+
+```python
+record: Record = ...
+app: App = ...
+
+feedback_result: FeedbackResult = f_qs_relevance.run(app=app, record=record)
+```
+
+The object can also be provided to an app wrapper for automatic evaluation:
+
+```python
+app: App = tru.Chain(...., feedbacks=[f_qs_relevance])
+```
+
+## Specifying Implementation Function and Aggregate
+
+The function or method provided to the `Feedback` constructor is the
+implementation of the feedback function which does the actual work of producing
+a float indicating some quantity of interest. Any callable can be provided here
+but there are additional requirements if your app uses the "deferred" feedback
+evaluation mode (when `feedback_mode=FeedbackMode.DEFERRED` are specified to app
+constructor). In those cases the callables must be methods that are globally
+importable (see the next section for details). The function/method performing
+the aggregation has the same requirements.
+
+The mapping of record/app contents to the arguments to the specified feedback
+implementation is provided via the `on_...` methods. These are discussed in a
+later section.
+
+### Global import requirement
+
+If using deferred evaluation, the feedback function implementations and
+aggregation implementations must be methods from a class that is globally
+importable. That is, the callables must be accessible were you to evaluate this
+code:
+
+```python
+import somepackage.[...].someclass 
+# [...] means optionally further package specifications
+
+provider = someclass(...) # constructor arguments can be included
+feedback_implementation = provider.somemethod
+```
+
+For provided feedback functions, `somepackage` is `trulens_eval.feedback` and
+`someclass` is `OpenAI` or one of the other `Provider` subclasses. Custom
+feedback functions likewise need to belong to a package that can be imported.
+Critically, classes defined locally in a notebook will not be importable this
+way.
+
+## Specifying Arguments
+
+The mapping between app/records to feedback implementation arguments is
+specified by the `on...` methods of the `Feedback` objects. The general form is:
+
+```python
+feedback: Feedback = feedback.on(argname1=selector1, argname2=selector2, ...)
+```
+
+That is, `Feedback.on(...)` returns a new `Feedback` object with additional
+argument mappings, the source of `argname1` is `selector1` and so on for further
+argument names. The types of `selector1` is `JSONPath` which we elaborate on in
+the further sections.
```

Comment:
nit: change

The types of `selector1` is `JSONPath` which we elaborate on in
the further sections.

to 

The types of `selector1` is `JSONPath` which will depend on if you are using langchain or llama-index. The respective sections will elaborate on usage.

Reply:
if you take some of my other edits; then then JSONPath descriptions should be coming up immediately next; so no need to reference "further section"